float32 ex
float32 ey
float32 BB_size
